### PR TITLE
Quadlet - add support for KubeDownForce

### PR DESCRIPTION
**Pull request by ygalblum**
Allow users to set --force for Stop command
Add doc and tests

#### Does this PR introduce a user-facing change?
Yes

```release-note
Quadlet - add support for setting --force for Stop command
```

### CODE REVIEW

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 1]
```diff
@@ -693,6 +694,10 @@ escaped to allow inclusion of whitespace and other control characters.
 
 This key can be listed multiple times.
 
+### `KubeDownForce=`
+
+Remove all resources when calling `podman kube down`.
```
Comment (vrothberg) [inline docs/source/markdown/podman-systemd.unit.5.md:699]:
```suggestion
Remove all resources including volumes when calling `podman kube down`.
```
?

**Review by vrothberg — APPROVED** [review]
LGTM
@edsantiago PTAL

**Comment by baude** [comment]
LGTM

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 2]
```diff
@@ -693,6 +694,10 @@ escaped to allow inclusion of whitespace and other control characters.
 
 This key can be listed multiple times.
 
+### `KubeDownForce=`
+
+Remove all resources including volumes when calling `podman kube down`.
```
Comment (edsantiago) [inline docs/source/markdown/podman-systemd.unit.5.md:699]: Missing commas: should be "...all resources, including volumes, when ...".

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

Want to wait for @edsantiago's head nod.

**Review by edsantiago — CHANGES_REQUESTED** [review]
One typo, few suggestions. Overall LGTM.

**Comment by rhatdan** [comment]
Is there a reason for this to not be default, to match up with the --force removal with .container? If this has been answered, then point me at it?  When would you not want the play to go down when stopping the service?

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 3]
```diff
@@ -637,6 +637,7 @@ Valid options for `[Kube]` are listed below:
 | ConfigMap=/tmp/config.map           | --config-map /tmp/config.map                                     |
 | ContainersConfModule=/etc/nvd\.conf | --module=/etc/nvd\.conf                                          |
 | GlobalArgs=--log-level=debug        | --log-level=debug                                                |
+| KubeDownForce=true                  | -force (for `podman kube down`)                                  |
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply docs/source/markdown/podman-systemd.unit.5.md]: Thanks, fixed

File: docs/source/markdown/podman-systemd.unit.5.md (hunk 2 above)
Comment (ygalblum) [reply docs/source/markdown/podman-systemd.unit.5.md]: Thanks, fixed

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 4]
```diff
@@ -693,6 +694,10 @@ escaped to allow inclusion of whitespace and other control characters.
 
 This key can be listed multiple times.
 
+### `KubeDownForce=`
+
+Remove all resources including volumes when calling `podman kube down`.
+Equivalent to the Podman `--force` option.
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply docs/source/markdown/podman-systemd.unit.5.md:700]: I understand your comment, but, phrasing is used all over this document. So, changing Podman->Command-Line should be done all over the man page

File: test/system/252-quadlet.bats [hunk 5]
```diff
@@ -1139,4 +1139,82 @@ EOF
...
+    run_podman container inspect --format "{{(index .Mounts 0).Type}}" $pod_name-$container_name
+    assert "$output" = "volume" \
+           "quadlet - kube oneshot: container should be attached to a volume of type volume"
+
+    run_podman container inspect --format "{{(index .Mounts 0).Name}}" $pod_name-$container_name
+    assert "$output" = "$quadlet_kube_volume_name" \
+           "quadlet - kube oneshot: container should be attached to the volume named $volume_name"
+
+    # Shutdown the service
+    service_cleanup $pod_service failed
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply test/system/252-quadlet.bats:1214]: Done

File: test/system/252-quadlet.bats [hunk 6]
```diff
@@ -1139,4 +1139,82 @@ EOF
...
+    run_podman 1 volume exists ${quadlet_kube_volume_name}
+
+    service_setup $pod_service
+
+    # Volume should exist
+    run_podman volume exists ${quadlet_kube_volume_name}
+
+    run_podman container inspect --format "{{(index .Mounts 0).Type}}" $pod_name-$container_name
+    assert "$output" = "volume" \
+           "quadlet - kube oneshot: container should be attached to a volume of type volume"
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply test/system/252-quadlet.bats]: Done

**Comment by ygalblum** [comment]
> Is there a reason for this to not be default, to match up with the --force removal with .container? If this has been answered, then point me at it? When would you not want the play to go down when stopping the service?

In the `.container` case, the removal is of the container and this is already the case for the containers started as part of a `.kube` file. The `--force` flag is not about containers, but about other resources and specifically volumes. 

Now, on top of this, I don't think we should break the current default behavior.

**Review by vrothberg — APPROVED** [review]
/lgtm

**Comment by vrothberg** [comment]
Isn't there an issue/RFE for this PR?

**Comment by ygalblum** [comment]
> Isn't there an issue/RFE for this PR?

The discussion for this requirement started here: #20284. But, this change does not cover all the requirements listed. So, I don't think this PR should close the issue.

**Comment by fsdrw08** [comment]
will this feature include in podman v4.8?

**Comment by ygalblum** [comment]
> will this feature include in podman v4.8?

Yes, it will

**Pull request merged**